### PR TITLE
Bug/team1 issue102 fix users/{userId}/tasks url

### PR DESCRIPTION
**Pull request by farhadbahrehmandhenry**
- issue #102 

- function URL updated to the original one after testing and making sure that it's working properly.

# how to use =>
- run the app
-- navigate to ui/react-ui
-- npm install
-- npm start

- see the supported endpoints on the Home page=>
Screen Shot 2021-02-02 at 10 01 08 AM

- Input the params and Click on the endpoint to trigger =>
-- you should receive an error if you leave the input empty =>
Screen Shot 2021-02-02 at 10 03 15 AM

-- also you should receive an error if your input has a wrong type (for example string instead of int) =>
Screen Shot 2021-02-02 at 10 05 59 AM

-- if the input is correct, you'll see the result => 
Screen Shot 2021-02-02 at 10 07 48 AM

- use the "Back" to go back to the home page

# spent time =>
about 6 hours

### CODE REVIEW

File: ui/react-ui/src/components/bootstrapTable/BootstrapTable.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ var BootstrapTable = (props) => {
 
   return (
     <div className='table-container'>
-      <Container>
+      <>
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/bootstrapTable/BootstrapTable.js:10]: Moving <Container> to another component - nice!

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js [hunk 2]
```diff
@@ -1,34 +1,48 @@
...
 
   var {userId} = useParams();
+  const history = useHistory();
 
   useEffect(() => {
-    // url should be replaced with => https://nsc-functionsapp-team1.azurewebsites.net/api/${userId}/tasks?
-    fetch(`https://w21httptriggernataliatest.azurewebsites.net/api/${userId}/tasks?`)
+    setUserTasks(['loading']);
+
+    fetch(`https://nsc-functionsapp-team1.azurewebsites.net/api/users/${userId}/tasks?`)
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/routes/userTasks/UserTasks.js:18]: Endpoint is updated with the correct url. Good work!

**Review by jak-ashuraliev — APPROVED** [review]
Updated and cleaned up - looks good!

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js (hunk 2 above)
Comment (David-Prism) [reply ui/react-ui/src/components/routes/userTasks/UserTasks.js:18]: ^Same comment from me!

Glad to know that the `tasks` endpoint is working. I just changed to the endpoint of this URL for the UserID endpoint, which is also working now.

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js [hunk 3]
```diff
@@ -1,34 +1,48 @@
...
+  var tasks;
+  var backHomeButtonStyle = {marginLeft: '1rem', marginBottom: '2rem'};
+  var backIcon = '<====';
+
+  if (userTasks[0] === 'loading') tasks = <Container>...loading</Container>;
+  else {
+    if (userTasks.length > 0) {
+      tasks = <BootstrapTable heatherItems={Object.keys(userTasks[0])} rows={userTasks} />;
+    }
+    else tasks = <h5>There is no task for this user</h5>;
```
Comment (David-Prism) [inline ui/react-ui/src/components/routes/userTasks/UserTasks.js:35]: I like this structure; it's much easier to read than the previous version. 
Did the old version work? Is the appearance (browser) the same as the new version?

**Review by David-Prism — APPROVED** [review]

File: ui/react-ui/src/components/routes/userTasks/UserTasks.js (hunk 3 above)
Comment (farhadbahrehmandhenry) [reply ui/react-ui/src/components/routes/userTasks/UserTasks.js:35]: yes, it worked, just changed it to be more readable.

File: ui/react-ui/package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "react-dom": "^17.0.1",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.1",
+    "react-tooltip": "^4.2.13",
```
Comment (jak-ashuraliev) [inline ui/react-ui/package.json:15]: I see you installed a new package for tooltips, which is good. We could also import bootstrap tooltips ;)

File: ui/react-ui/src/components/bootstrapBackButton/BootstrapBackButton.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {useHistory} from "react-router-dom";
+import Button from 'react-bootstrap/Button';
+
+var BackButton = () => {
+  const history = useHistory();
+
+  return (
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/bootstrapBackButton/BootstrapBackButton.js:8]: Created a new reusable BackButton component - good work!

File: ui/react-ui/src/components/routes/home/Home.js [hunk 6]
```diff
@@ -1,20 +1,115 @@
...
+        {key: '/users', params: []}, 
+        {key: '/users/{userId}', params: [{key: 'userId', value: 0, type: 'number', regex: /{userId}/g}]},
+        {key: '/users/{userId}/tasks', params: [{key: 'userId', value: 0, type: 'number', regex: /{userId}/g}]}
+      ]
+    },
+    badInputAlert: '',
+    emptyInputAlert: ''
+  };
+
+  handleParamInputChange({method, endpointKey, paramKey, value}) {
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/routes/home/Home.js:23]: Good implementation of handling params and validation.

File: ui/react-ui/src/components/routes/home/Home.js [hunk 7]
```diff
@@ -1,20 +1,115 @@
...
+  render() {
+    var {endpointsByMethods, badInputAlert, emptyInputAlert} = this.state;
+    var methods = Object.keys(endpointsByMethods);
+ 
+    // what happens here is basically mapping over endpointsByMethods and rendering a method title, endpoints, and params
+    return (
+      <Container className="text-center mt-5">
+        <h2 className='main-header'>Supported Endpoints:</h2>
+        <hr className='main-horizontal-rule'></hr>
+        {badInputAlert && <Alert variant="danger" onClose={() => this.setState({badInputAlert: ''})} dismissible>{badInputAlert}</Alert>}
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/routes/home/Home.js:78]: The bootstrap alert is implemented - consistency = success!

File: ui/react-ui/src/components/routes/users/Users.js [hunk 8]
```diff
@@ -25,29 +26,32 @@ export default class Users extends Component {
...
-							<td>{user.firstName}</td>
-							<td>{user.lastName}</td>
-							<td>{user.email}</td>
-							</tr>
-						</tbody>
-						))}
-					</Table>
-				</Container>
+        <>
+          <BackButton />
```
Comment (jak-ashuraliev) [inline ui/react-ui/src/components/routes/users/Users.js:30]: Thanks for creating a back button on the Users component. The man!

**Review by jak-ashuraliev — APPROVED** [review]
Overall - good cleanup, good implementation of bootstrap components like Button, Alerts, Tables. Good work @farhadbahrehmandhenry !

File: ui/react-ui/package.json (hunk 4 above)
Comment (farhadbahrehmandhenry) [reply ui/react-ui/package.json:15]: I used bootstrap tooltip first, but it has some dependencies that apparently are deprecated and caused me to get a bunch of errors, so I moved to this one.

**Pull request merged**